src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessorTest.java

```diff
@@ -65,25 +72,62 @@ public void should_read_from_blob_storage_and_save_metadata_in_database_when_zip
         processor.processJurisdiction("SSCS");
 
         // then
-        Envelope dbEnvelope = envelopeRepository.findAll().get(0);
+        List<Envelope> dbEnvelopes = envelopeRepository.findAll();
 
-        assertThat(dbEnvelope.getStatus()).isEqualTo(PROCESSED);
-        assertThat(dbEnvelope.getScannableItems())
+        assertThat(dbEnvelopes)
+            .hasSize(1)
+            .extracting("status")
+            .containsOnlyOnce(PROCESSED);
+        assertThat(dbEnvelopes.get(0).getScannableItems())
             .extracting("documentUrl")
             .hasSameElementsAs(ImmutableList.of(DOCUMENT_URL1, DOCUMENT_URL2));
 
         // and
-        List<ProcessEvent> processEvents = processEventRepository.findAll();
-        assertThat(processEvents).hasSize(3);
-
         String failureReason = "Document metadata not found for file " + pdf1.getFilename();
 
-        assertThat(processEvents)
+        assertThat(processEventRepository.findAll())
+            .hasSize(3)
             .extracting("container", "zipFileName", "event", "reason")
             .containsOnly(
                 tuple(testContainer.getName(), ZIP_FILE_NAME_SUCCESS, DOC_UPLOAD_FAILURE, failureReason),
                 tuple(testContainer.getName(), ZIP_FILE_NAME_SUCCESS, DOC_UPLOADED, null),
                 tuple(testContainer.getName(), ZIP_FILE_NAME_SUCCESS, DOC_PROCESSED, null)
             );
     }
+
+    @Test
+    public void should_fail_to_upload_pdfs_when_retrying_with_reupload_task() throws Exception {
+        // given
+        uploadZipToBlobStore(ZIP_FILE_NAME_SUCCESS); //Zip file with metadata and pdfs
+
+        byte[] test1PdfBytes = toByteArray(getResource("1111001.pdf"));
+        byte[] test2PdfBytes = toByteArray(getResource("1111002.pdf"));
+
+        Pdf pdf1 = new Pdf("1111001.pdf", test1PdfBytes);
+        Pdf pdf2 = new Pdf("1111002.pdf", test2PdfBytes);
+
+        given(documentManagementService.uploadDocuments(ImmutableList.of(pdf1, pdf2)))
+            .willReturn(Collections.emptyMap());
```

Comment:
Better to throw exception rather than returning empty response ?

Reply:
Not making much of a difference, but for purposes of having different upload failure origin - can do that. So both are checked